main.go

```diff
@@ -9,16 +35,42 @@ const (
 )
 
 func main() {
-	//logger, err := bootstrap.SetupLogger(defaultLogLevel)
-	//if err != nil {
-	//	log.Printf("error creating zap logger %v", err)
-	//	os.Exit(1)
-	//}
-
-	//ctx := app.SetupSignalHandler(logger, func(signal os.Signal) {
-	//
-	//})
-
-	// app.Setup()
-	//app.Start()
+	args := os.Args[1:]
+	checkArgs(args)
+	logger, err := bootstrap.SetupLogger(defaultLogLevel)
+	if err != nil {
+		log.Fatalf("error creating zap logger %v", err)
+	}
+
+	ctx, cancelFn := signal.SetupHandler(logger, bootstrap.CaptureExitCode, types.DefaultExitCodeFilePath)
+
+	// Add flags
+	fs := flag.CommandLine
+	cmd.EtcdCmd.AddFlags(fs)
+	_ = fs.Parse(args[1:])
+
+	// Print all flags
+	printFlags(logger)
+
+	// InitAndStartEtcd command
+	if err = cmd.EtcdCmd.Run(ctx, cancelFn, logger); err != nil {
+		logger.Fatal("error during start or run of etcd", zap.Error(err))
+	}
+}
+
+// Should check if any arg is help and print
+func checkArgs(args []string) {
+	//check if any unsupported command is specified. Print help if that is the case
+	if len(args) < 1 || !cmd.IsCommandSupported(args[0]) {
+		_ = cmd.PrintHelp(os.Stderr)
```

Comment:
please handle this error as well ? you can use `log.Fatalf` as next line you are calling `os.Exit()`

Reply:
There is no need to handle any error here. The only code flow if the control reaches here is to exit the app.